### ISSUE TITLE
Docker : nginx crash and  database "secmonkey" does not exist

### ISSUE THREAD

**Issue by raccoon4me** [Netflix/security_monkey#637]
Hi, 

I followed this tutorial:

https://github.com/Netflix/security_monkey/blob/develop/docs/docker.rst

```
docker-compose build
```
```
Successfully built ca0e5f3fd8ff
```
```
docker-compose up -d postgres
```
```
Status: Downloaded newer image for postgres:9
Creating secmonkey-db
```

```
docker-compose up -d init
```
```
secmonkey-db is up-to-date
Creating init
```

But when i run 
```
docker-compose up
```
```
...
secmonkey-nginx | Sat Mar 25 09:25:18 UTC 2017 Error: Missing files required for SSL
secmonkey-nginx | sed: cannot rename /etc/nginx/conf.d/securitymonkey.conf: Device or resource busy
secmonkey-nginx | 2017/03/25 09:25:18 [emerg] 1#1: PEM_read_bio_X509_AUX("/etc/nginx/ssl/server.crt") failed (SSL: error:0906D06C:PEM routines:PEM_read_bio:no start line:Expecting: TRUSTED CERTIFICATE)
secmonkey-nginx | nginx: [emerg] PEM_read_bio_X509_AUX("/etc/nginx/ssl/server.crt") failed (SSL: error:0906D06C:PEM routines:PEM_read_bio:no start line:Expecting: TRUSTED CERTIFICATE)
...
secmonkey-nginx exited with code 1
...
secmonkey-db | FATAL:  database "secmonkey" does not exist
secmonkey-scheduler | 2017-03-25 09:25:19,590 WARNING: Scheduler Exception: (OperationalError) FATAL:  database "secmonkey" does not exist
secmonkey-scheduler |  None None [in /usr/local/src/security_monkey/security_monkey/scheduler.py:231]
secmonkey-scheduler | 2017-03-25 09:25:19,595 WARNING: Traceback (most recent call last):
secmonkey-scheduler |   File "/usr/local/src/security_monkey/security_monkey/scheduler.py", line 208, in setup_scheduler
secmonkey-scheduler |     accounts = Account.query.filter(Account.third_party == False).filter(Account.active == True).all()  # noqa
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2279, in all
secmonkey-scheduler |     return list(self)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2391, in __iter__
secmonkey-scheduler |     return self._execute_and_instances(context)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2404, in _execute_and_instances
secmonkey-scheduler |     close_with_result=True)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2395, in _connection_from_session
secmonkey-scheduler |     **kw)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 843, in connection
secmonkey-scheduler |     close_with_result=close_with_result)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 847, in _connection_for_bind
secmonkey-scheduler |     return self.transaction._connection_for_bind(engine)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 315, in _connection_for_bind
secmonkey-scheduler |     conn = bind.contextual_connect()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1698, in contextual_connect
secmonkey-scheduler |     self.pool.connect(),
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 324, in connect
secmonkey-scheduler |     return _ConnectionFairy._checkout(self)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 608, in _checkout
secmonkey-scheduler |     fairy = _ConnectionRecord.checkout(pool)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 425, in checkout
secmonkey-scheduler |     rec = pool._do_get()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 928, in _do_get
secmonkey-scheduler |     return self._create_connection()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 277, in _create_connection
secmonkey-scheduler |     return _ConnectionRecord(self)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 396, in __init__
secmonkey-scheduler |     self.connection = self.__connect()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 509, in __connect
secmonkey-scheduler |     connection = self.__pool._creator()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/strategies.py", line 95, in connect
secmonkey-scheduler |     connection_invalidated=invalidated
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 185, in raise_from_cause
secmonkey-scheduler |     reraise(type(exception), exception, tb=exc_tb)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/strategies.py", line 89, in connect
secmonkey-scheduler |     return dialect.connect(*cargs, **cparams)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 366, in connect
secmonkey-scheduler |     return self.dbapi.connect(*cargs, **cparams)
secmonkey-scheduler |   File "/usr/local/lib/python2.7/dist-packages/psycopg2-2.6.2-py2.7-linux-x86_64.egg/psycopg2/__init__.py", line 164, in connect
secmonkey-scheduler |     conn = _connect(dsn, connection_factory=connection_factory, async=async)
secmonkey-scheduler | OperationalError: (OperationalError) FATAL:  database "secmonkey" does not exist
secmonkey-scheduler |  None None
secmonkey-scheduler |  [in /usr/local/src/security_monkey/security_monkey/scheduler.py:232]
secmonkey-scheduler | 2017-03-25 09:25:19,595 DEBUG: Logging exception from scheduler with location: None to the database. [in /usr/local/src/security_monkey/security_monkey/datastore.py:696]
secmonkey-db | FATAL:  database "secmonkey" does not exist
secmonkey-scheduler | 2017-03-25 09:25:19,598 ERROR: Encountered exception while logging exception to database: [in /usr/local/src/security_monkey/security_monkey/datastore.py:730]
secmonkey-scheduler | 2017-03-25 09:25:19,598 ERROR: (OperationalError) FATAL:  database "secmonkey" does not exist
secmonkey-scheduler |  None None [in /usr/local/src/security_monkey/security_monkey/datastore.py:731]
secmonkey-scheduler | Traceback (most recent call last):
secmonkey-scheduler |   File "/usr/local/src/security_monkey/security_monkey/datastore.py", line 726, in store_exception
secmonkey-scheduler |     db.session.commit()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/scoping.py", line 149, in do
secmonkey-scheduler |     return getattr(self.registry(), name)(*args, **kwargs)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 765, in commit
secmonkey-scheduler |     self.transaction.commit()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 370, in commit
secmonkey-scheduler |     self._prepare_impl()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 350, in _prepare_impl
secmonkey-scheduler |     self.session.flush()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 1879, in flush
secmonkey-scheduler |     self._flush(objects)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 1997, in _flush
secmonkey-scheduler |     transaction.rollback(_capture_exception=True)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/util/langhelpers.py", line 57, in __exit__
secmonkey-scheduler |     compat.reraise(exc_type, exc_value, exc_tb)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 1961, in _flush
secmonkey-scheduler |     flush_context.execute()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/unitofwork.py", line 370, in execute
secmonkey-scheduler |     rec.execute(self)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/unitofwork.py", line 523, in execute
secmonkey-scheduler |     uow
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/persistence.py", line 45, in save_obj
secmonkey-scheduler |     uowtransaction)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/persistence.py", line 140, in _organize_states_for_save
secmonkey-scheduler |     states):
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/persistence.py", line 818, in _connections_for_states
secmonkey-scheduler |     base_mapper)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 228, in connection
secmonkey-scheduler |     return self._connection_for_bind(bind)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 304, in _connection_for_bind
secmonkey-scheduler |     conn = self._parent._connection_for_bind(bind)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py", line 315, in _connection_for_bind
secmonkey-scheduler |     conn = bind.contextual_connect()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1698, in contextual_connect
secmonkey-scheduler |     self.pool.connect(),
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 324, in connect
secmonkey-scheduler |     return _ConnectionFairy._checkout(self)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 608, in _checkout
secmonkey-scheduler |     fairy = _ConnectionRecord.checkout(pool)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 425, in checkout
secmonkey-scheduler |     rec = pool._do_get()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 928, in _do_get
secmonkey-scheduler |     return self._create_connection()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 277, in _create_connection
secmonkey-scheduler |     return _ConnectionRecord(self)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 396, in __init__
secmonkey-scheduler |     self.connection = self.__connect()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/pool.py", line 509, in __connect
secmonkey-scheduler |     connection = self.__pool._creator()
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/strategies.py", line 95, in connect
secmonkey-scheduler |     connection_invalidated=invalidated
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 185, in raise_from_cause
secmonkey-scheduler |     reraise(type(exception), exception, tb=exc_tb)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/strategies.py", line 89, in connect
secmonkey-scheduler |     return dialect.connect(*cargs, **cparams)
secmonkey-scheduler |   File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 366, in connect
secmonkey-scheduler |     return self.dbapi.connect(*cargs, **cparams)
secmonkey-scheduler |   File "/usr/local/lib/python2.7/dist-packages/psycopg2-2.6.2-py2.7-linux-x86_64.egg/psycopg2/__init__.py", line 164, in connect
secmonkey-scheduler |     conn = _connect(dsn, connection_factory=connection_factory, async=async)
secmonkey-scheduler | OperationalError: (OperationalError) FATAL:  database "secmonkey" does not exist
secmonkey-scheduler |  None None
secmonkey-scheduler | 2017-03-25 09:25:19,600 INFO: Scheduler started [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:574]
secmonkey-scheduler | INFO:apscheduler.scheduler:Scheduler has been shut down
secmonkey-scheduler | 2017-03-25 09:25:19,600 INFO: Scheduler has been shut down [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:606]
secmonkey-scheduler exited with code 0
```

For the nginx part, the problem is that server.crt and server.key are not file, but directories
can be fixed by either adding your own cert or following this closed issue:

https://github.com/Netflix/security_monkey/issues/589

```
drwxr-xr-x 2 root root 4096 Mar 25 09:25 server.crt
drwxr-xr-x 2 root root 4096 Mar 25 09:25 server.key
```

Any chance we could get this issue resolved please.

Thanks a lot and keep your hard work ;)

**Comment by monkeysecurity** [comment]
Hey @raccoon4me - It'd be a huge help if you could send a PR.  Could we have the nginx disable ssl if certs are not provided?

**Comment by monkeysecurity** [comment]
fixed in #657
